src/ui/verificationGutterStatusView.ts

```diff
@@ -261,28 +261,28 @@ export default class VerificationGutterStatusView {
   private async getRangesOfLineStatus(params: IVerificationGutterStatusParams): Promise<Map<LineVerificationStatus, Range[]>> {
     const perLineStatus = this.addCosmetics(params.perLineStatus);
 
-    const symbolParams = await this.symbolStatusView.getFirstStatusForCurrentVersion(params.uri);
-    const linesToSkip = new Set(symbolParams.namedVerifiables.map(v => v.nameRange.start.line));
+    const symbolParams = (await this.symbolStatusView?.getFirstStatusForCurrentVersion(params.uri)) ?? [];
+    const linesToSkip = [ -1, ...symbolParams.map(testItem => testItem.range!.start.line).sort((a, b) => a - b), perLineStatus.length ];
```

Comment:
Since you are changing substantially this method. could you please
1) Refactor the rest of the code after this line in a public static helper method named `PerLineStatusToRanges`, taking `perLineStatus` and `linesToSkip`  as arguments (and prepend `-1` to `lineToSkip`)
2) At the end of `src/test/extensions.test.ts`, add the following

```javascript
suite("Verification Gutter", () => {
  test("PerLineStatusToRanges", () => {
    var ranges= VerificationGutterStatusView.PerLineStatusToRanges([1, 1, 1, 3, 3, 3, 2, 2, 2], [2, 4, 6]);
    assert.strictEqual(3, result.size);
    assert.deepStrictEqual([new Range(0, 1, 1, 1)], ranges.get(1));
    assert.deepStrictEqual([new Range(7, 1, 7, 1)], ranges.get(2));
    assert.deepStrictEqual([new Range(3, 1, 3, 1), new Range(5, 1, 5, 1)], ranges.get(3));
  })
});
```

3) Run the tests to verify that it works.